### PR TITLE
11199 add upsell link button component

### PR DESCRIPTION
**Pull request by Xyfi**
## Summary

This PR can be summarized in the following changelog entry:

* Adds the `UpsellLinkButton` component.

## Test instructions

This PR can be tested by following these steps:

* Test in combination with Yoast/wordpress-seo#11231. See this PR's test instructions.

Fixes Yoast/wordpress-seo#11199

### CODE REVIEW

File: composites/Plugin/Shared/components/UpsellLinkButton.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	&:hover,
+	&:focus,
+	&:active {
+		color: ${ colors.$color_black };
+		background: ${ colors.$color_button_upsell_hover };
+	}
+	
+	:active {
+		position: relative;
+  		top: 2px;
```
Comment (afercia) [inline composites/Plugin/Shared/components/UpsellLinkButton.js]: this line has mixed spaces and tabs

**Comment by afercia** [comment]
Ideally, the new (link)button should be added to the standalone app "Buttons" tab for testing, maintenance, and discoverability. Will address during CR.

File: composites/Plugin/Shared/components/UpsellLinkButton.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	white-space: nowrap;
+	display: inline-flex;
+	border-radius: 4px;
+	background-color: ${ colors.$color_button_upsell };
+	padding: 4px 8px 8px;
+	box-shadow: inset 0 -4px 0 rgba(0, 0, 0, 0.2);
+	border: none;
+	text-decoration: none;
+	font-size: inherit;
+
```
Comment (afercia) [inline composites/Plugin/Shared/components/UpsellLinkButton.js]: there's a tab character to remove in this empty line

File: composites/Plugin/Shared/components/UpsellLinkButton.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	text-decoration: none;
+	font-size: inherit;
+	
+	&:hover,
+	&:focus,
+	&:active {
+		color: ${ colors.$color_black };
+		background: ${ colors.$color_button_upsell_hover };
+	}
+
```
Comment (afercia) [inline composites/Plugin/Shared/components/UpsellLinkButton.js]: there's a tab character to remove in this empty line

**Comment by afercia** [comment]
CR 👍

**Comment by andizer** [comment]
Acceptance 👍

**Pull request merged**